Esri/arcgis-maps-sdk-swift-samples
# [New] Show device location history

## Description

This PR implements `Show device location history` in `Maps` category.

URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Ting/ShowLocationHistory/Shared/Samples/Show%20device%20location%20history

## Linked Issue(s)

- `common-samples/issues/3820`

## How To Test

Run the sample and see if the track always works, and the objects get deallocated correctly when the sample quits.

## Screenshots

![show-device-location-history](https://user-images.githubusercontent.com/9660181/218149922-4bb01b18-880f-4a42-ad7a-f8419ffb65f3.png)

## To Discuss

1. Do we need to weakly capture self in the standalone tasks?
2. Make sure the `for await newLocation in locationDisplay.dataSource.locations` section is correct. This sample's key part is  to demonstrate async stream is the equivalent of `locationChangedHandler` pattern.
